### PR TITLE
ADR: Define supported installation method

### PR DESCRIPTION
**Pull request by balloob**
ADR to define what is a supported installation method.

As introduced earlier in https://www.home-assistant.io/blog/2020/05/26/installation-methods-and-community-guides-wiki/

### CODE REVIEW

File: adr/0012-define-supported-installation-method.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+So as Home Assistant grows and evolves, some of these installation guides became outdated and stopped working. We wouldn’t even know it was broken until a user raised an issue. But when they do, we wouldn’t know how to fix it unless we could get a hold of the original contributor.
+
+This can be frustrating. Any guide on our official website should lead to a working system. A system that not only works today, but also tomorrow.
+
+## Decision
+
+A supported installation method in the Home Assistant context means:
+
+> A way of installing and running Home Assistant One, Home Assistant Container, Home Assistant Supervised or Home Assistant Core; in a way that is supported by the Home Assistant developers. Supported means the installation method is tested and documented in the official documentation. Running Home Assistant using such a supported method, leads to the optimal user experience.
```
Comment (Adminiuga) [inline adr/0012-define-supported-installation-method.md:25]: > A way of installing and running Home Assistant One, Home Assistant Container, Home Assistant Supervised or Home Assistant Core

What is "Home Assistant One" ?

Comment (balloob) [reply adr/0012-define-supported-installation-method.md:25]: 😆 

That is for the next ADR, but that will be the new name of referring to the installation method of what is currently Home Assistant. 

The idea behind it is that it is confusing when people have issues. What are you running? "Home Assistant". "Ok, but what HA?"

Comment (Adminiuga) [reply adr/0012-define-supported-installation-method.md:25]: One to  rule them all :)

Comment (DubhAd) [inline adr/0012-define-supported-installation-method.md:25]: > Home Assistant One
is this the latest name for the thing previously called Hass.io?

Comment (thomasloven) [inline adr/0012-define-supported-installation-method.md:25]: I think this introduces another question.
What is *installed* and what is the *method*?

As I see it, at the base there's Home Assistant - the python program.
Then there's different ways of making that run
- One
- in a container
- in a container that's controlled by the supervisor
- standalone

That would make One, Container, Supervised and - in a way - Core *installation methods*, which makes this a circular definition.

Comment (balloob) [reply adr/0012-define-supported-installation-method.md]: Every method installs core, which is what is the home automation part. But how it is installed and managed is part of the installation method, which is what we're defining/naming.

**Pull request merged**